File: Native/Test/Node/mr-array-map.js
FC = require("Fabric").createClient();

cv = FC.MR.createConstValue("Size", 10);

ago = FC.KLC.createArrayGeneratorOperator("foo.kl", "operator foo(io Scalar output, Size index) { output = Scalar(index); }", "foo");
ag = FC.MR.createArrayGenerator(cv, ago);

mo = FC.KLC.createArrayMapOperator("bar.kl", "operator bar(Scalar input, io Scalar output) { output = input * input; }", "bar");
m = FC.MR.createArrayMap(ag, mo);

var count = m.getCount();
console.log("count = " + count);
for (var i=0; i<count; ++i)
  console.log("produce("+i+") = "+m.produce(i));

FC.close();

File: Native/Test/Python/mr-array-map.py
#
#  Copyright 2010-2012 Fabric Technologies Inc. All rights reserved.
#

import fabric

client = fabric.createClient()

cv = client.MR.createConstValue("Size", 10)

ago = client.KLC.createArrayGeneratorOperator("foo.kl", "operator foo(io Scalar output, Size index) { output = Scalar(index); }", "foo")
ag = client.MR.createArrayGenerator(cv, ago)

mo = client.KLC.createArrayMapOperator("bar.kl", "operator bar(Scalar input, io Scalar output) { output = input * input; }", "bar")
m = client.MR.createArrayMap(ag, mo)

count = m.getCount()
print("count = " + str(count))
for i in range(0, count):
  print("produce("+str(i)+") = "+str(m.produce(i)))

client.close()
